Orange/data/table.py

```diff
@@ -912,15 +912,16 @@ def is_copy(self):
 
     def ensure_copy(self):
         """
-        Ensure that the table owns its data; copy arrays when necessary
+        Ensure that the table owns its data; copy arrays when necessary.
+        The check is skipped for sparse matrices since they don't have views like numpy arrays.
         """
-        if self.X.base is not None:
+        if not sp.issparse(self.X) and self.X.base is not None:
             self.X = self.X.copy()
```

Comment:
Currently not, this is left to the constructor. As far as I understand `ensure_copy` only ensures that the data is not a view and makes a copy if it is. Since sparse matrices do not have views (at least I think so) I propose to skip the check. Currently the copy is made in the constructor anyway and hence we probably do not need to make an other here.